File: app/src/main/java/com/jakubwyc/mysecretvault/presenter/MainPresenter.java
package com.jakubwyc.mysecretvault.presenter;

import com.jakubwyc.mysecretvault.presenter.Presenter;
import com.jakubwyc.mysecretvault.view.MainView;

public class MainPresenter implements Presenter<MainView> {

    private MainView view;

    @Override
    public void attachView(MainView view) {
        this.view = view;
    }

    @Override
    public void detachView() {
        view = null;
    }
}

File: app/src/main/kotlin/com/jakubwyc/mysecretvault/main/MainPresenter.kt
package com.jakubwyc.mysecretvault.main

import com.jakubwyc.mysecretvault.Presenter

class MainPresenter : Presenter<MainView> {

    private var view: MainView? = null

    override fun attachView(view: MainView) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }
}
